distributed/tests/test_steal.py

```diff
@@ -1870,3 +1871,63 @@ async def test_trivial_workload_should_not_cause_work_stealing(c, s, *workers):
     await c.gather(futs)
     events = s.events["stealing"]
     assert len(events) == 0
+
+
+@gen_cluster(
+    client=True,
+    nthreads=[("", 1)] * 2,
+    config={
+        "distributed.scheduler.worker-saturation": 1.5,
+        "distributed.scheduler.work-stealing": False,
+    },
+    timeout=2,
+)
+async def test_transition_queued_memory(c, s, a, b):
+    """https://github.com/dask/distributed/issues/7200"""
+    ev0 = Event()
+    ev1 = Event()
+    sched_id = id(s)
+
+    async def f0(ev):
+        await ev.wait()
+        s = next(si for si in Scheduler._instances if id(si) == sched_id)
+        # Note that the task is async, so it runs in the same thread as the scheduler
+        s.reschedule("x-0", stimulus_id="steal")
```

Comment:
As pointed out in https://github.com/dask/distributed/issues/7200#issuecomment-1293416880
this is not how stealing works and using `Scheduler.reschedule` in this way is not intended use. If this is the only way to provoke this transition, we should remove `Scheduler.reschedule` (or rather make it private, users should raise the `Reschedule` exception.